web/META-INF/resources/js/controllers.js

```diff
@@ -0,0 +1,261 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you 
+ * may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/* Controllers */
+var mudrodControllers = angular.module('mudrodControllers', []);
+
+mudrodControllers.controller('searchCtrl', ['$scope', '$rootScope', '$location', 'Autocomplete', 'SearchOptions',
+    function ($scope, $rootScope, $location, Autocomplete, SearchOptions) {
+        $scope.options = {
+            opt: 'Or'
+        };
+
+        $scope.complete = function (string) {
+            $scope.hidethis = false;
+            Autocomplete.get({term: string},
+                function success(response) {
+                    $scope.filterSearch = response;
+
+                },
+                function error(errorResponse) {
+                    console.log("Error:" + JSON.stringify(errorResponse));
+                }
+            );
+        };
+
+        $scope.fillTextbox = function (string) {
+            $scope.options.query = string;
+            $scope.hidethis = true;
+        };
+
+        $scope.search = function (options) {
+            $rootScope.searchOptions = angular.copy(options);
+            $location.path("/metadataView/" + options.query + '/' + options.opt);
+        };
+
+        $scope.$watch(function () {
+            return SearchOptions.getSearchOptions();
+        }, function (newValue, oldValue) {
+            if (newValue !== null) {
+                $scope.options.query = newValue.query;
+                $scope.options.opt = newValue.opt;
+            }
+        }, true);
+    }]);
+
+mudrodControllers.controller('vocabularyCtrl', ['$scope', '$rootScope', 'VocabList',
+    function vocabularyCtrl($scope, $rootScope, VocabList) {
+
+        var word = $rootScope.searchOptions.query;
+        var opt = $rootScope.searchOptions.opt;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "opt" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of the unused 'opt' variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AUnusedVariable)